react/LocationSearch.tsx

```diff
@@ -46,6 +56,29 @@ const useDebouncedValue = (value: string, delayInMs: number) => {
   return debouncedValue
 }
 
+const useSuggestions = (
+  searchTerm: string
+): { suggestions: AddressSuggestion[]; loading: boolean } => {
+  const [executeSuggestAddresses, { data, error, loading }] = useLazyQuery<
+    Query,
+    QuerySuggestAddressesArgs
+  >(SUGGEST_ADDRESSES)
+
+  useEffect(() => {
+    if (searchTerm.trim().length) {
+      executeSuggestAddresses({ variables: { searchTerm } })
+    }
+  }, [searchTerm, executeSuggestAddresses])
+
+  useEffect(() => {
+    if (error) {
+      console.error(error.message)
+    }
+  }, [error])
+
+  return { suggestions: data?.suggestAddresses ?? [], loading }
```

Comment:
wdyt of returning an array? a la `useState`

Reply:
I was going to do that, but couldn't think on a strong reason, so went with the most readable version imo. Do you know any advantages of returning an array?

Reply:
you don't have to rename the object properties on the callsite:

```tsx
const [suggestions, areSuggestionsLoading] = useSuggestions(searchTerm)
const [myNewSuggestions] = useSuggestions(anotherSearchTerm)
```

Reply:
fair enough, will change